### PR TITLE
Allow metadata flag to be specified multiple times.

### PR DESCRIPTION
**Pull request by ehankland**

### CODE REVIEW

**Comment by cmccoy** [comment]
Will this support comma-separated arguments, or is it breaking?

**Comment by ehankland** [comment]
This breaks the old behavior

**Comment by voellm** [comment]
Is there a way to allow the following....

--metadata=foo1:bar1,foo2,bar2 --metadata foo3:bar3 ?



On Mon, Nov 30, 2015 at 1:01 PM, ehankland <notifications@github.com> wrote:

> This breaks the old behavior
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/684#issuecomment-160760467>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/
Benchmark like a pro... PerfKitBenchmarker
<https://github.com/GoogleCloudPlatform/PerfKitBenchmarker>

**Comment by ehankland** [comment]
The newest version supports both comma separated lists and specifying `--metadata` multiple times

File: perfkitbenchmarker/publisher.py [hunk 1]
```diff
@@ -76,12 +76,11 @@
...
 
-flags.DEFINE_list(
+flags.DEFINE_multistring(
     'metadata',
     [],
-    'A list of key-value pairs that will be added to the labels field of all '
-    'samples as metadata. Each key-value pair in the list should be colon '
-    'separated.')
+    'A colon separated key-value pair that will be added to the labels field '
+    'of all samples as metadata. This option can be repeated multiple times.')
```
Comment (cmccoy) [inline perfkitbenchmarker/publisher.py]: Could you note that kv-pairs can be comma-separated within a `--metadata` specification?

**Comment by cmccoy** [comment]
LGTM.

File: perfkitbenchmarker/publisher.py (hunk 1 above)
Comment (ehankland) [inline perfkitbenchmarker/publisher.py]: Done

**Pull request merged**